R/db-odbc-mssql.R

```diff
@@ -68,7 +68,28 @@
       as.double     = sql_cast("NUMERIC"),
       as.character  = sql_cast("VARCHAR(MAX)"),
       log           = sql_prefix("LOG"),
+      nchar         = sql_prefix("LEN"),
+      atan2         = sql_prefix("ATN2"),
+      substr        = sql_prefix("SUBSTRING"),
+      ceil          = sql_prefix("CEILING"),
       ceiling       = sql_prefix("CEILING"),
+      # Casting as BIT converts the Integer result into an actual logical
```

Comment:
It doesn't look like it would.  I tried:

```
Select CAST((mpg IS NULL) AS BIT)from mtcars 
```

```
Error in new_result(connection@ptr, statement) : 
  nanodbc/nanodbc.cpp:1344: 42000: Incorrect syntax near the keyword 'AS'., Statement(s) could not be prepared.,  
Failed to execute SQL chunk
```

This  query worked:

```
Select CAST(mpg AS BIT)from mtcars 
```
So I tried a different evaluation call inside a parenthesis, but it also fails: 

```
Select CAST((mpg > 23) AS BIT)from mtcars 
```

```
Error in new_result(connection@ptr, statement) : 
  nanodbc/nanodbc.cpp:1344: 42000: Incorrect syntax near the keyword 'AS'., Statement(s) could not be prepared.,  
Failed to execute SQL chunk
```